pkg/project/config.go

```diff
@@ -42,16 +45,86 @@ type HandlerConfig struct {
 // }
 
 type BaseConfig struct {
-	Name     string `yaml:"name"`
-	Dir      string `yaml:"-"`
-	Handlers []any  `yaml:"handlers"`
+	Name       string         `yaml:"name"`
+	Dir        string         `yaml:"-"`
+	Handlers   []any          `yaml:"handlers"`
+	Containers []DockerConfig `yaml:"containers" validate:"handlers_containers_required,validate_docker_object,dive"`
+}
+
+type DockerConfig struct {
+	Dockerfile string            `yaml:"dockerfile"`
+	Image      string            `yaml:"image" validate:"omitempty,validate_docker_image"`
+	Context    string            `yaml:"context"`
+	Args       map[string]string `yaml:"args"`
+	Nitric     bool              `yaml:"nitric"`
```

Comment:
Can you add BaseComputeConfig here for configurable typing (will also need to be specific as `inline`.